backend/src/main/java/gov/cdc/usds/simplereport/service/TestOrderService.java

```diff
@@ -235,17 +236,23 @@ public TestOrder getTestOrder(Organization org, UUID id) {
   @Deprecated // switch to specifying device-specimen combo
   public TestOrder editQueueItem(
       UUID testOrderId, String deviceId, String result, Date dateTested) {
-    TestOrder order = this.getTestOrder(testOrderId);
+    try {
+      lockOrder(testOrderId);
 
-    if (deviceId != null) {
-      order.setDeviceSpecimen(_dts.getDefaultForDeviceId(deviceId));
-    }
+      TestOrder order = this.getTestOrder(testOrderId);
+
+      if (deviceId != null) {
+        order.setDeviceSpecimen(_dts.getDefaultForDeviceId(deviceId));
+      }
 
-    order.setResult(result == null ? null : TestResult.valueOf(result));
+      order.setResult(result == null ? null : TestResult.valueOf(result));
 
-    order.setDateTestedBackdate(dateTested);
+      order.setDateTestedBackdate(dateTested);
 
-    return _repo.save(order);
+      return _repo.save(order);
+    } finally {
+      unlockOrder(testOrderId);
```

Comment:
I think this will be called even when the lock isn't acquired. I think `lockOrder`  for `editQueueItem` and `AddTestResult` should occur before the try finally block

Reply:
Ack, you're right! Thanks for this. Updated in 57e2ee